### ISSUE TITLE
Suggestion: List of common easyblocks

### ISSUE THREAD

**Issue by verdurin** [easybuilders/easybuild-docs#164]
A page listing the features of common easyblocks and how to decide when to use them would be helpful.
I'm thinking especially of common queries such as picking `Tarball` versus `PackedBinary` versus `Binary`.

If this is already there, perhaps better links are needed?

**Comment by boegel** [comment]
I agree we could have way better documentation on these generic easyblocks, definitely also included very common ones like `ConfigureMake`, `CMakeMake`, `PythonPackage` + `PythonBundle`, etc.

The recently added `Cargo` easyblock should also be covered, since that has some nice feature that are a bit atypical.

Any specific suggestions for where to park these pages, and how to organize them? One landing page with some high-level overview (like `PythonPackage` vs `PythonBundle`, `Tarball` vs `Binary`), and then specific pages for specific generic easyblocks?
